test/e2e/e2e_test.go

```diff
@@ -0,0 +1,189 @@
+package e2e
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/G-Research/yunikorn-history-server/cmd/yunikorn-history-server/commands"
+	"github.com/G-Research/yunikorn-history-server/internal/health"
+	"github.com/G-Research/yunikorn-history-server/internal/yunikorn/model"
+	"github.com/G-Research/yunikorn-history-server/test/k8s"
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/utils/ptr"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	testNamespacePrefix = "yunikorn-e2e-"
+)
+
+func TestE2E(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping e2e test in short mode")
+	}
+	
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	ns := createTestNamespace(ctx, t, k8s.GetTestK8sClient(t))
+	t.Cleanup(func() {
+		deleteTestNamespace(context.Background(), t, k8s.GetTestK8sClient(t), ns)
+	})
+
+	serverURL := "http://localhost:8989"
+	go runApp(ctx)
+
+	assert.Eventually(t, func() bool {
+		healthy, err := getReadinessStatus(serverURL)
+		return healthy && err == nil
+	}, 10*time.Second, 500*time.Millisecond)
+
+	// sleep for 2 seconds just in case so all goroutines are ready
+	time.Sleep(2 * time.Second)
+
+	k8sClient := k8s.GetTestK8sClient(t)
+
+	job := testJob()
+	_, err := k8sClient.BatchV1().Jobs(ns).Create(ctx, job, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("error creating test job: %v", err)
+	}
+
+	expectedCounts := model.EventTypeCounts{
+		"APP-ADD":          7,
+		"APP-REMOVE":       7,
+		"APP-SET":          5,
+		"NODE-ADD":         3,
+		"NODE-REMOVE":      3,
+		"USERGROUP-ADD":    3,
+		"USERGROUP-REMOVE": 3,
+		"QUEUE-ADD":        2,
+		"QUEUE-REMOVE":     2,
+	}
+	assert.Eventually(t, func() bool {
```

Comment:
I am getting an error in my IDE for this code block. May be we can take a look at it one more time.

<img width="1043" alt="image" src="https://github.com/G-Research/yunikorn-history-server/assets/32765701/06bf6301-0986-428f-8a33-01e9f161152c">

Reply:
This needs to be merged first as it is a requirement for this PR - https://github.com/G-Research/yunikorn-history-server/pull/76